app/code/Magento/Payment/Model/Method/Adapter.php

```diff
@@ -508,11 +508,12 @@ private function executeCommand($commandCode, array $arguments = [])
         }
 
         if (isset($arguments['payment'])) {
+            $payment = $arguments['payment'];
             $arguments['payment'] = $this->paymentDataObjectFactory->create($arguments['payment']);
-        }
-
-        if ($this->commandExecutor !== null) {
-            return $this->commandExecutor->executeByCode($commandCode, $arguments);
+            
+            if ($this->commandExecutor !== null) {
```

Comment:
Ya,I can see its the case - the last commit should pass a null in case the payment is not in $arguments